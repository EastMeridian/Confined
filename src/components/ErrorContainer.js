import React from 'react';
import t from 'prop-types';
import ErrorMessage from './ErrorMessage';

const ErrorContainer = ({ error, children, ...other }) => (
  <>
    {error && <ErrorMessage>{error}</ErrorMessage>}
    {!error && React.cloneElement(children, other)}
  </>
);

ErrorContainer.propTypes = {
  error: t.string,
  children: t.node,
};

ErrorContainer.defaultProps = {
  error: null,
  children: null,
};

export default ErrorContainer;
