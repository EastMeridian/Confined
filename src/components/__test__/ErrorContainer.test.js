import '@testing-library/jest-dom';
import { create } from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import ErrorContainer from '../ErrorContainer';
import '@testing-library/jest-dom/extend-expect';

describe('ErrorContainer component', () => {
  const testMessage = 'Test children';
  const TestChildren = () => <div>{testMessage}</div>;
  const testError = 'Error Message';

  it('should matches the snapshot', () => {
    const container = create(<ErrorContainer error={testError} />);
    expect(container.toJSON()).toMatchSnapshot();
  });


  it('should display children with no error provided', () => {
    const { getByText } = render(<ErrorContainer><TestChildren /></ErrorContainer>);
    expect(getByText(testMessage)).toBeInTheDocument();
  });

  it('should display error but not childen with error', () => {
    const { getByText, queryByText } = render(
      <ErrorContainer error={testError}><TestChildren /></ErrorContainer>,
    );
    expect(getByText(testError)).toBeInTheDocument();
    expect(queryByText(testMessage)).not.toBeInTheDocument();
  });
});
