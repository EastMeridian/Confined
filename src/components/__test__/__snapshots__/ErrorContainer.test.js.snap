// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorContainer component should matches the snapshot 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <img
    alt="empty_result"
    src="test"
    style={
      Object {
        "filter": "blur(2px)",
        "objectFit": "contain",
        "position": "absolute",
        "width": "100%",
      }
    }
  />
  <span
    style={
      Object {
        "color": "white",
        "fontFamily": "Roboto",
        "position": "absolute",
      }
    }
  >
    Error Message
  </span>
</div>
`;
